lib/widgets/action_sheet.dart

```diff
@@ -0,0 +1,33 @@
+import 'package:flutter/material.dart';
+import 'package:share_plus/share_plus.dart';
+
+import '../api/model/model.dart';
+import 'draggable_scrollable_modal_bottom_sheet.dart';
+
+void showMessageActionSheet({required BuildContext context, required Message message}) {
+  showDraggableScrollableModalBottomSheet(
+    context: context,
+    builder: (BuildContext context) {
+      return Column(
+        children: [
+          MenuItemButton(
+            leadingIcon: Icon(Icons.adaptive.share),
+            onPressed: () async {
+              // Close the message action sheet; we're about to show the share
+              // sheet. (We could do this after the sharing Future settles, but
+              // on iOS I get impatient with how slowly our action sheet
+              // dismisses in that case.)
+              Navigator.of(context).pop();
+
+              // TODO: to support iPads, we're asked to give a
+              // `sharePositionOrigin` param, or risk crashing / hanging:
+              //   https://pub.dev/packages/share_plus#ipad
```

Comment:
Wacky. That is definitely below the quality standards that would apply to a first-party plugin.

Upstream threads:
https://github.com/fluttercommunity/plus_plugins/issues/870#issuecomment-1141550034
https://github.com/fluttercommunity/plus_plugins/issues/305#issuecomment-874910403

And it sounds like the old first-party plugin it replaced didn't have this problem:
https://github.com/fluttercommunity/plus_plugins/issues/305#issuecomment-874801852

Well, the TODO is good enough for merging for the prototype.